[bug] React component failed to build with Next.js 12

@giscus

## What happened?

When I update `giscus-react` from `2.2.2` to `2.2.3`, Next.js 12 building crashed with an error:

```text
info  - Collecting page data .Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'D:\Work\Community\KaiYuanShe\OWS\node_modules\.pnpm\@giscus+react@2.2.3_sfoxds7t5ydpegc3knd667wn6m\node_modules\react\jsx-runtime' imported from D:\Work\Community\KaiYuanShe\OWS\node_modules\.pnpm\@giscus+react@2.2.3_sfoxds7t5ydpegc3knd667wn6m\node_modules\@giscus\react\dist\index.mjs    
Did you mean to import react@17.0.2/node_modules/react/jsx-runtime.js?
    at new NodeError (node:internal/errors:387:5)
    at finalizeResolution (node:internal/modules/esm/resolve:330:11)
    at moduleResolve (node:internal/modules/esm/resolve:907:10)
    at defaultResolve (node:internal/modules/esm/resolve:1115:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:837:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

> Build error occurred
Error: Failed to collect page data for /article/[id]
    at D:\Work\Community\KaiYuanShe\OWS\node_modules\.pnpm\next@12.3.3_sfoxds7t5ydpegc3knd667wn6m\node_modules\next\dist\build\utils.js:916:15 {
  type: 'Error'
}
```

## Environment

| Software         | Version(s)          |
| ---------------- | ------------------- |
| React            | 17.0.2              |
| Next.js          | 12.3.3              |
| Node             | 16.18.0             |
| PNPM             | 7.15.0              |
| Operating System | Windows 10 Pro 21H2 |

--- a/react/package.json
+++ b/react/package.json
@@ -1,14 +1,14 @@
 {
   "name": "@giscus/react",
   "version": "2.2.3",
-  "main": "dist/index.cjs",
-  "module": "dist/index.mjs",
+  "main": "dist/index.js",
+  "module": "dist/wrapper.mjs",
   "types": "dist/index.d.ts",
   "exports": {
     ".": {
       "types": "./dist/index.d.ts",
-      "require": "./dist/index.cjs",
-      "import": "./dist/index.mjs"
+      "require": "./dist/index.js",
+      "import": "./dist/wrapper.mjs"
     }
   },
   "files": [

--- a/react/vite.config.ts
+++ b/react/vite.config.ts
@@ -7,15 +7,12 @@ const resolvePath = (str: string) => resolve(__dirname, str);
 
 // https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [react()],
+  plugins: [react(), wrapper()],
   build: {
     lib: {
       entry: resolve(__dirname, 'src/lib/index.ts'),
-      formats: ['cjs', 'es'],
-      fileName: (format) => ({
-        cjs: 'index.cjs',
-        es: 'index.mjs',
-      }[format]),
+      formats: ['cjs'],
+      fileName: 'index',
     },
     rollupOptions: {
       external: ['react', 'react-dom', 'react/jsx-runtime'],
@@ -37,3 +34,16 @@ export default defineConfig({
     },
   },
 });
+
+function wrapper() {
+  return {
+    name: 'wrapper',
+    generateBundle() {
+      this.emitFile({
+        type: 'asset',
+        fileName: 'wrapper.mjs',
+        source: `import module from './index.js';\n\nexport default module;`,
+      });
+    },
+  };
+}

